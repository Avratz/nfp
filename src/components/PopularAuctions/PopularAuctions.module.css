.popularAuctions {
  margin-top: 4rem;
  margin-bottom: 4rem;
  display: flex;
  flex-direction: column;
  gap: 2rem;
}

.image {
  aspect-ratio: 2 / 3;
  position: relative;
  border-radius: 1rem;
  overflow: hidden;
}

.auctionData {
  display: flex;
  flex-direction: column;
  gap: 2.5rem;
}

.authorAndPrice {
  display: flex;
  justify-content: space-between;
  margin-top: 1.25rem;
}

.box {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.priceIcon {
  width: 40px;
  height: 40px;
  position: relative;
  border-radius: 50%;
  background-color: var(--green);
}

.title {
  letter-spacing: -1px;
}

.topTitle {
  color: var(--neutral-4);
  font-family: var(--font-poppins);
  font-weight: var(--font-regular);
  font-size: var(--text-xs);
  line-height: var(--line-xs);
}

.bottomTitle {
  font-family: var(--font-poppins) !important;
  font-weight: var(--font-medium) !important;
}

.auctionButtons {
  display: flex;
  flex-direction: column;
  width: 100%;
  gap: 0.5rem;
}

.navigation {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 0.5rem;
}

.arrow {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0;
  border: 2px solid var(--neutral-3);
  border-radius: 50%;
  width: 40px;
  height: 40px;
}

.disabled {
  border: none;
  pointer-events: none;
}

.boxData {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
}

@media screen and (min-width: 768px) {
  .popularAuctions {
    margin-top: 8rem;
    flex-direction: row;
    align-items: flex-start;
  }
  .image {
    width: 55%;
  }

  .auctionData {
    width: 45%;
  }
  .navigation {
    justify-content: flex-start;
  }
}

@media screen and (min-width: 768px) {
  .popularAuctions {
    gap: 8rem;
    margin-bottom: 8rem;
  }
  .image {
    width: 65%;
    max-height: 800px;
  }

  .auctionData {
    width: 35%;
  }
}

@media screen and (min-width: 1920px) {
  .popularAuctions {
    gap: 12rem;
  }
  .image {
    width: 60%;
  }
  .auctionData {
    width: 40%;
  }
  .auctionStatus {
    padding: 3rem 0;
  }
}
