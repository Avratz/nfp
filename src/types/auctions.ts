export type Auction = {
  id: number
  instantPrice: string
  highestBid: string
  author: string
  authorAvatar: string
  type: string
  stock: number
  likes: number
  createdAt: Date
  endsAt: Date
  media: {
    id: number
    image: string
    image2x: string
  }
  attributes: {
    id: number
    color: string
    type: string
  }
  bidUsers: [
    {
      id: number
      name: string
      avatar: string
    },
  ]
}
