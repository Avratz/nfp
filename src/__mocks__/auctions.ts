import { Auction } from '@/types/auctions'

export const allAuctions: Auction[] = [
  {
    id: 10,
    instantPrice: '4.45 ETH',
    highestBid: '0.010 ETH',
    author: 'Fernando Medina',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
    type: 'Art',
    stock: 1,
    likes: 2,
    createdAt: '2022-06-05T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 4,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121559/NFPAISANOS/img_125_mrxfbh.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121559/NFPAISANOS/img_125_mrxfbh.png',
    },
    attributes: {
      id: 4,
      color: 'black',
      type: 'epic',
    },
    bidUsers: [
      {
        id: 6,
        name: 'Sofia Coolen',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/05_rdi4su.png',
      },
      {
        id: 7,
        name: 'Martin Acosta',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
      },
    ],
  },
  {
    id: 11,
    instantPrice: '2.45 ETH',
    highestBid: '0.100 ETH',
    author: 'Martin Acosta',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/06_mjddoi.png',
    type: 'Art',
    stock: 3,
    likes: 10,
    createdAt: '2022-06-06T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 5,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121554/NFPAISANOS/img_131_d7wmyk.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121559/NFPAISANOS/img_125_mrxfbh.png',
    },
    attributes: {
      id: 5,
      color: 'pink',
      type: 'rare',
    },
    bidUsers: [
      {
        id: 8,
        name: 'Nicolas Sieiro',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/02_nqwu5e.png',
      },
      {
        id: 9,
        name: 'Carla Luna',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/03_ifygi0.png',
      },
    ],
  },
  {
    id: 12,
    instantPrice: '2.15 ETH',
    highestBid: '0.500 ETH',
    author: 'Santiago Lopez',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
    type: 'Photography',
    stock: 5,
    likes: 10,
    createdAt: '2022-06-07T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 6,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121551/NFPAISANOS/img_126_ae6zew.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121551/NFPAISANOS/img_126_ae6zew.png',
    },
    attributes: {
      id: 6,
      color: 'orange',
      type: 'uncommon',
    },
    bidUsers: [
      {
        id: 10,
        name: 'Martin Acosta',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
      {
        id: 11,
        name: 'Antonella Richieri',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/08_fsbhry.png',
      },
    ],
  },
  {
    id: 13,
    instantPrice: '10.05 ETH',
    highestBid: '1.500 ETH',
    author: 'Santiago Echazu',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/01_qgg62f.png',
    type: 'Art',
    stock: 1,
    likes: 30,
    createdAt: '2022-06-10T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 7,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121542/NFPAISANOS/img_101_y2iczn.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121542/NFPAISANOS/img_101_y2iczn.png',
    },
    attributes: {
      id: 7,
      color: 'orange',
      type: 'legendary',
    },
    bidUsers: [
      {
        id: 12,
        name: 'Martin Acosta',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
      {
        id: 13,
        name: 'Antonella Richieri',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/08_fsbhry.png',
      },
      {
        id: 14,
        name: 'Antonella Richieri',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/08_fsbhry.png',
      },
      {
        id: 15,
        name: 'Martin Acosta',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
    ],
  },
  {
    id: 14,
    instantPrice: '1.05 ETH',
    highestBid: '0.40 ETH',
    author: 'Alvaro Echazu',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/02_nqwu5e.png',
    type: 'Art',
    stock: 1,
    likes: 400,
    createdAt: '2022-06-15T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 8,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121536/NFPAISANOS/img_102_a9rkb7.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121536/NFPAISANOS/img_102_a9rkb7.png',
    },
    attributes: {
      id: 8,
      color: 'black',
      type: 'legendary',
    },
    bidUsers: [
      {
        id: 16,
        name: 'Akira Kaneto',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
      },
      {
        id: 17,
        name: 'Facundo Aguinaga',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
      },
    ],
  },
  {
    id: 15,
    instantPrice: '1.30 ETH',
    highestBid: '0.022 ETH',
    author: 'Sofia Coolen',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/03_ifygi0.png',
    type: 'Photography',
    stock: 2,
    likes: 14,
    createdAt: '2022-06-20T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 9,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_115_lolex4.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121534/NFPAISANOS/img_115_lolex4.png',
    },
    attributes: {
      id: 9,
      color: 'blue',
      type: 'uncommon',
    },
    bidUsers: [
      {
        id: 18,
        name: 'Julian Krupka',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/01_qgg62f.png',
      },
      {
        id: 19,
        name: 'Julian Alvarez',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
      },
    ],
  },
  {
    id: 16,
    instantPrice: '6.52 ETH',
    highestBid: '0.031 ETH',
    author: 'Flora Quiroga',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/08_fsbhry.png',
    type: 'Photography',
    stock: 1,
    likes: 201,
    createdAt: '2022-06-30T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 10,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_119_o3ouhw.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_119_o3ouhw.png',
    },
    attributes: {
      id: 10,
      color: 'pink',
      type: 'legendary',
    },
    bidUsers: [
      {
        id: 20,
        name: 'Manuel Torres',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
      },
      {
        id: 21,
        name: 'Franco Kaus',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/02_nqwu5e.png',
      },
    ],
  },
  {
    id: 17,
    instantPrice: '0.002 ETH',
    highestBid: '0.00005 ETH',
    author: 'Mar√≠a Del Mar Gestoso',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/03_ifygi0.png',
    type: 'Art',
    stock: 1,
    likes: 0,
    createdAt: '2022-06-01T11:10:33.000Z',
    endsAt: '2022-09-06T11:42:33.000Z',
    media: {
      id: 11,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_133_slwzia.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121534/NFPAISANOS/img_133_slwzia.png',
    },
    attributes: {
      id: 11,
      color: 'red',
      type: 'uncommon',
    },
    bidUsers: [],
  },
  {
    id: 18,
    instantPrice: '0.028 ETH',
    highestBid: '0.00052 ETH',
    author: 'Julian Cerino',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/01_qgg62f.png',
    type: 'Art',
    stock: 10,
    likes: 5,
    createdAt: '2022-06-14T11:10:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 12,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_114_d5usms.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121534/NFPAISANOS/img_114_d5usms.png',
    },
    attributes: {
      id: 12,
      color: 'black',
      type: 'rare',
    },
    bidUsers: [
      {
        id: 22,
        name: 'Leo Suarez',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
    ],
  },
  {
    id: 19,
    instantPrice: '1.00 ETH',
    highestBid: '0.012 ETH',
    author: 'Martin Russo',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/06_mjddoi.png',
    type: 'Art',
    stock: 1,
    likes: 30,
    createdAt: '2022-06-10T11:10:33.000Z',
    endsAt: '2022-07-06T11:42:33.000Z',
    media: {
      id: 13,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121532/NFPAISANOS/img_122_s2bk3s.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121532/NFPAISANOS/img_122_s2bk3s.png',
    },
    attributes: {
      id: 13,
      color: 'black',
      type: 'legendary',
    },
    bidUsers: [
      {
        id: 23,
        name: 'Juan Lazo',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
      {
        id: 24,
        name: 'Exequiel Germano',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/02_nqwu5e.png',
      },
    ],
  },
  {
    id: 20,
    instantPrice: '3.00 ETH',
    highestBid: '0.21 ETH',
    author: 'Belen Lago',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/05_rdi4su.png',
    type: 'Photography',
    stock: 5,
    likes: 15,
    createdAt: '2022-06-03T11:10:33.000Z',
    endsAt: '2022-07-06T11:42:33.000Z',
    media: {
      id: 14,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_104_zklzrv.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121534/NFPAISANOS/img_104_zklzrv.png',
    },
    attributes: {
      id: 14,
      color: 'orange',
      type: 'rare',
    },
    bidUsers: [
      {
        id: 25,
        name: 'Nacho Margulies',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
    ],
  },
]
