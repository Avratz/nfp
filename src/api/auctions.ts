import { getCachedOneMinute } from '@/lib/fetcher'
import { Auction } from '@/types/auctions'

export function getPopularAuctions(): Promise<Auction[]> {
  return getCachedOneMinute('/popular', process.env.API_KEY as string)
}

export function getAllAuctions(): Promise<Auction[]> {
  return getCachedOneMinute('/aunctions', process.env.API_KEY as string)
}
