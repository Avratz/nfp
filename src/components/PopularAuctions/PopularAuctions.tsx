'use client'
import Image from 'next/image'

import { cx } from 'class-variance-authority'
import { motion } from 'framer-motion'

// this is overkill, but it's an example of how to use '@tanstack/react-query'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

import { useCarousel } from '@/hooks/useCarousel'
import { Auction } from '@/types/auctions'
import { Heading } from '@/ui/Heading/Heading'
import { Avatar } from '@/ui/Avatar/Avatar'
import { Button } from '@/ui/Button/Button'
import { Icon } from '@/ui/Icon/Icon'
import { Container } from '@/ui/Container/Container'

import { PopularAuctionsStatus } from '@/components/PopularAuctionsStatus/PopularAuctionsStatus'
import { PopularAuctionsActionButtons } from '@/components/PopularAuctionsActionButtons/PopularAuctionsActionButtons'

import styles from './PopularAuctions.module.css'

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      cacheTime: 0,
    },
  },
})

let animationDefault = {
  initial: {
    x: 0,
    y: 50,
    opacity: 0,
  },
  whileInView: {
    opacity: 1,
  },
  animate: {
    x: 0,
    y: 0,
  },
  transition: {
    duration: 0.3,
    ease: 'easeIn',
  },
}

export function PopularAuctions({ data }: { data: [Auction[]] }) {
  const [auctions] = data
  const { selectedSlide, changeSlide, currentIndex } =
    useCarousel<Auction[]>(auctions)

  return (
    <Container as="section" className={styles.popularAuctions}>
      <motion.picture className={styles.image} {...animationDefault}>
        <Image
          src={selectedSlide.media.image}
          alt=""
          fill
          sizes="(max-width: 768px) 100vw,
              50vw"
          style={{
            objectFit: 'cover',
          }}
          priority
        />
      </motion.picture>

      <motion.div className={styles.auctionData} {...animationDefault}>
        <div>
          <Heading as="h3" size="6xl" className={styles.title}>
            The {getLastName(selectedSlide.author)} Network Â®
          </Heading>
          <div className={styles.authorAndPrice}>
            <div className={styles.box}>
              <Avatar
                src={selectedSlide.authorAvatar}
                alt={selectedSlide.author}
              />
              <div className={styles.boxData}>
                <div className={styles.topTitle}>Creator</div>
                <Heading as="h4" size="sm" className={styles.bottomTitle}>
                  {selectedSlide.author}
                </Heading>
              </div>
            </div>
            <div className={styles.box}>
              <Button
                className={styles.priceIcon}
                intent="icon"
                title="Buy now"
              >
                <Icon name="stop" />
              </Button>
              <div className={styles.boxData}>
                <div className={styles.topTitle}>Instant price</div>
                <Heading as="h4" size="sm" className={styles.bottomTitle}>
                  {selectedSlide.instantPrice}
                </Heading>
              </div>
            </div>
          </div>
        </div>

        <QueryClientProvider client={queryClient}>
          <PopularAuctionsStatus selectedSlide={selectedSlide} />
        </QueryClientProvider>

        <PopularAuctionsActionButtons highestBid={selectedSlide.highestBid} />

        <div className={styles.navigation}>
          <Button
            intent="icon"
            className={cx(
              styles.arrow,
              currentIndex === 0 ? styles.disabled : '',
            )}
            title="Previous Auction"
            onClick={changeSlide('prev')}
          >
            <Icon name="arrowLeft" />
          </Button>
          <Button
            intent="icon"
            className={cx(
              styles.arrow,
              currentIndex === auctions.length - 1 ? styles.disabled : '',
            )}
            title="Next Auction"
            onClick={changeSlide('next')}
          >
            <Icon name="arrowRight" />
          </Button>
        </div>
      </motion.div>
    </Container>
  )
}

function getLastName(name: string) {
  // name is formatted as 'First Last'
  return name.split(' ').pop()
}
