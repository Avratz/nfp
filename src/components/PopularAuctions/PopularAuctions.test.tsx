import { render } from '@testing-library/react'
import { PopularAuctions } from './PopularAuctions'

import { allAuctions } from '@/__mocks__/auctions'

jest.mock('@/ui/Icon/Icon', () => {
  function Icon() {
    return <div></div>
  }

  Icon.displayName = 'Icon'
  return {
    __esModule: true,
    Icon,
  }
})

const firstThreeAuctions = allAuctions.slice(0, 3)
describe('PopularAuctions', () => {
  it('renders PopularAuctions', () => {
    let { container } = render(<PopularAuctions data={[firstThreeAuctions]} />)
    expect(container).toMatchSnapshot()
  })
})
