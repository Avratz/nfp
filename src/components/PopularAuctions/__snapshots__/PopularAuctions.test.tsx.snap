// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularAuctions renders PopularAuctions 1`] = `
<div>
  <section
    class="popularAuctions container containerPadding"
  >
    <picture
      class="image"
      style="opacity: 0; transform: translateX(0px) translateY(50px) translateZ(0);"
    >
      <img
        alt=""
        data-nimg="fill"
        decoding="async"
        sizes="(max-width: 768px) 100vw, 50vw"
        src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fc_scale%2Cw_1000%2Fv1657121559%2FNFPAISANOS%2Fimg_125_mrxfbh.png&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fc_scale%2Cw_1000%2Fv1657121559%2FNFPAISANOS%2Fimg_125_mrxfbh.png&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fc_scale%2Cw_1000%2Fv1657121559%2FNFPAISANOS%2Fimg_125_mrxfbh.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fc_scale%2Cw_1000%2Fv1657121559%2FNFPAISANOS%2Fimg_125_mrxfbh.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fc_scale%2Cw_1000%2Fv1657121559%2FNFPAISANOS%2Fimg_125_mrxfbh.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fc_scale%2Cw_1000%2Fv1657121559%2FNFPAISANOS%2Fimg_125_mrxfbh.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fc_scale%2Cw_1000%2Fv1657121559%2FNFPAISANOS%2Fimg_125_mrxfbh.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fc_scale%2Cw_1000%2Fv1657121559%2FNFPAISANOS%2Fimg_125_mrxfbh.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fc_scale%2Cw_1000%2Fv1657121559%2FNFPAISANOS%2Fimg_125_mrxfbh.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fc_scale%2Cw_1000%2Fv1657121559%2FNFPAISANOS%2Fimg_125_mrxfbh.png&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </picture>
    <div
      class="auctionData"
      style="opacity: 0; transform: translateX(0px) translateY(50px) translateZ(0);"
    >
      <div>
        <h3
          class="heading 6xl title"
        >
          The 
          Medina
           Network Â®
        </h3>
        <div
          class="authorAndPrice"
        >
          <div
            class="box"
          >
            <picture
              class="avatar base"
            >
              <img
                alt="Fernando Medina"
                data-nimg="1"
                decoding="async"
                height="36"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fv1657153929%2FNFPAISANOS%2F07_xrihyb.png&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fv1657153929%2FNFPAISANOS%2F07_xrihyb.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fv1657153929%2FNFPAISANOS%2F07_xrihyb.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fv1657153929%2FNFPAISANOS%2F07_xrihyb.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fv1657153929%2FNFPAISANOS%2F07_xrihyb.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fv1657153929%2FNFPAISANOS%2F07_xrihyb.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fv1657153929%2FNFPAISANOS%2F07_xrihyb.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fv1657153929%2FNFPAISANOS%2F07_xrihyb.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdvmll0ruo%2Fimage%2Fupload%2Fv1657153929%2FNFPAISANOS%2F07_xrihyb.png&w=3840&q=75 3840w"
                style="color: transparent;"
                width="36"
              />
            </picture>
            <div
              class="boxData"
            >
              <div
                class="topTitle"
              >
                Creator
              </div>
              <h4
                class="heading sm bottomTitle"
              >
                Fernando Medina
              </h4>
            </div>
          </div>
          <div
            class="box"
          >
            <button
              class="button icon base priceIcon"
              title="Buy now"
            >
              <div />
            </button>
            <div
              class="boxData"
            >
              <div
                class="topTitle"
              >
                Instant price
              </div>
              <h4
                class="heading sm bottomTitle"
              >
                4.45 ETH
              </h4>
            </div>
          </div>
        </div>
      </div>
      <div
        class="auctionStatus"
      >
        <div>
          <div
            class="auctionStatusTitle"
          >
            Current Bid
          </div>
          <h4
            class="heading 5xl"
          >
            0.010 ETH
          </h4>
          <div
            class="skeleton"
          />
        </div>
        <div>
          <div
            class="auctionStatusTitle"
          >
            Auction ending in
          </div>
          <div
            class="timer"
          >
            <div
              class="time"
            >
              <div
                class="timeValue"
              >
                00
              </div>
              <div
                class="timeUnit"
              >
                Hrs
              </div>
            </div>
            <div
              class="time"
            >
              <div
                class="timeValue"
              >
                00
              </div>
              <div
                class="timeUnit"
              >
                mins
              </div>
            </div>
            <div
              class="time"
            >
              <div
                class="timeValue"
              >
                00
              </div>
              <div
                class="timeUnit"
              >
                secs
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="auctionButtons"
      >
        <button
          class="button accent base"
        >
          Place a bid
        </button>
        <button
          class="button primary base"
        >
          View item
        </button>
      </div>
      <div
        class="navigation"
      >
        <button
          class="button icon base arrow disabled"
          title="Previous Auction"
        >
          <div />
        </button>
        <button
          class="button icon base arrow"
          title="Next Auction"
        >
          <div />
        </button>
      </div>
    </div>
  </section>
</div>
`;
